agents/loco_mc_agent.py

```diff
@@ -344,6 +361,29 @@ def maybe_dump_memory_to_dashboard(self):
             }
             sio.emit("memoryState", self.dashboard_memory["db"])
 
+    def register_hook(self, receiver, sender):
+        """
+        allows for registering hooks using the event dispatcher
+        """
+        allowed = [self.perceive,]
+        if sender in allowed:
+            self._dispatch_signal.connect(receiver, sender)
+        else:
+            raise ValueError("Unknown hook event {}. Available options are: {}".format(sender.__name__, allowed))
```

Comment:
would this line need `[a.__name__ for a in allowed]` instead of `allowed`?

Reply:
Oh sorry, I must have misinterpreted your last comment. Yeah that's a much cleaner of way of doing it, I'll push those changes asap!